webviews/welcome-page/src/util/useVscode.ts

```diff
@@ -0,0 +1,36 @@
+import type { VsCodeApi } from 'vscode-messenger-webview';
+import { useVscodeApi } from './VscodeApiProvider';
+
+export const useVscode = () => {
+  const vscodeApi = useVscodeApi();
+
+  const openUrl = (url: string) => {
+    vscodeApi.vscode.postMessage({ type: 'open-external-link', url });
+  };
+
+  const executeCommand = (command: string) => {
+    vscodeApi.vscode.postMessage({ type: 'execute-command', command });
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const getData = (callback: (value: any) => void) => {
+    window.addEventListener('message', event => {
+      callback(event.data);
+      console.log(event.data);
+    });
+  };
```

Comment:
eventlisteners like this should be attached via an useEffect and cleanup in it's return.
I'm also not understand why this needs to live in this useVscode hook... 
see other comment